prow/github/client.go

```diff
@@ -786,6 +802,34 @@ func (c *Client) CloseIssue(org, repo string, number int) error {
 	return err
 }
 
+// StateCannotBeChanged represents the "custom" GitHub API
+// error that occurs when a resource cannot be changed
+type StateCannotBeChanged struct {
+	Message string
+}
+
+func (s StateCannotBeChanged) Error() string {
+	return s.Message
+}
+
+// StateCannotBeChanged implements error
+var _ error = (*StateCannotBeChanged)(nil)
+
+// convert to a StateCannotBeChanged if appropriate or else return the original error
+func stateCannotBeChangedOrOriginalError(err error) error {
+	requestErr, ok := err.(requestError)
+	if ok && len(requestErr.Errors) > 0 {
+		for _, subErr := range requestErr.Errors {
+			if strings.Contains(subErr.Message, stateCannotBeChangedMessagePrefix) {
```

Comment:
This will break anytime GitHub changes their messages. Instead, we should be relying on the status code, which should be more reliable.